### DISCUSSION TITLE
Is it possible to send an ICAL event directly without the recipient having to open the ICS file?

### DISCUSSION THREAD

**Discussion by PurwadiPw** [spatie/icalendar-generator#81] [Q&A] [answered]
I had been looking at an .ics file which contains an Outlook event. I then rebuilt this in PHP and added all the necessary information such as the start/end date, subject, content, location, etc.

The generated file is then stored on the server for a short time and then sent as an email until I delete it again.

Currently, this generated .ics file is then sent to someone in my network and as soon as he opens the file, the appointment/event is then entered directly.

Theoretically it is also possible that someone other than the intended recipient opens it and as long as the intended recipient is entered in the .ics file, the intended recipient gets the event in his calendar.

But now I want these events to be entered automatically without the recipient having to click on the .ics file first.

My question is, is there a way to do this?

By changing the .ics file, possible settings in Outlook or the possibility to execute files on the server via Laravel commands?

Thanks in advance

**Comment by rubenvanassche** [comment, marked as answer]
Hi @PurwadiPw ,

The Ical standard does not describe such thing, so I don't think it is possible. It depends on the calendar/email client.

**Reply by PurwadiPw** [reply]
finally i ask my clients to subscribe to a calendar url i have created then i make an update on that calendar feed which will then appear automatically.

**Comment by husky83** [comment]
I know exactly what you want to get, I have been struggling with it for a long time, and it is certainly possible because, for example, Zoom, Webex or other applications can send such an invitation.
Unfortunately, because of Laravel himself (notification via e-mail or simply an e-mail created by Laravel) I cannot do it.
From pure PHP I managed to do it after using PHPMailer.
There, it is enough to add to the e-mail:


`$mail = new PHPMailer(true);     
$mail->CharSet = 'UTF-8';
$mail->Encoding = 'base64';
$mail->SMTPDebug = 0;
$mail->isSMTP();
$mail->Host = env('MAIL_HOST'); 
$mail->SMTPAuth = env('MAIL_HOST');
$mail->Username = env('MAIL_USERNAME', null);  
$mail->Password =  env('MAIL_PASSWORD', null);       
$mail->SMTPSecure = env('MAIL_ENCRYPTION', 'tls'); 
$mail->Port = env('MAIL_PORT', 2525);
$mail->ContentType = 'text/calendar'; 
$mail->setFrom(config('mail.from.address'), config('mail.from.name'));
$mail->addAddress($adressee);
$mail->addReplyTo(config('mail.from.address'), config('mail.from.name'));
$mail->Body = $mailBody;
$mail->Subject =  $mailSubject;
$mail->isHTML(true);
$mail->AltBody = $cal->get();
$mail->Ical = $cal->get();
$mail->send()`

Most important is:
`$mail->Ical = $cal->get();
$mail->AltBody = $cal->get();`


Unfortunately, I don't know why, but then the CSS doesn't look like it should, but the calendar is automatically added.
Therefore, in the case of notifications, you can always add a new PHPMailer-based channel, and in the case of regular messages, create a class somewhere on the side that will do what you need to ...
